skargo/src/skargo.sk

```diff
@@ -112,17 +117,29 @@ private fun findUserConfig(optPath: ?String = None()): UserConfig {
   config
 }
 
-private fun run(args: Array<String>, verbose: Bool): void {
-  if (verbose) {
-    // TODO: colors
-    print_error(`>> ${args.join(" ")}`)
-  };
-  p = System.subprocess(args);
-  if (verbose || !p.success()) {
-    print_raw(p.stdout);
-    print_error_raw(p.stderr);
+private fun run(args: Array<String>, console: Console): void {
+  console.status_verbose("Running", args.join(" "));
+  (stdout_cb, stderr_cb) = if (console.verbosity >= VerbosityExtraVerbose()) {
+    (print_error_raw, print_raw)
+  } else {
+    ((_) -> void, (_) -> void)
   };
+
+  p = System.subprocess(args, stdout_cb, stderr_cb);
   if (!p.success()) {
+    console.error(
+      "command exited with non zero status\n\n" +
+        "Caused by:\n" +
+        "  process didn't exit successfully: \`" +
```

Comment:
Maybe doesn't matter, but I tend to avoid printing quotes to shell output if it is easy
```suggestion
        "  process did not exit successfully: \`" +
```